tests/filecheck/parser-printer/operation_signature.mlir

```diff
@@ -0,0 +1,62 @@
+// RUN: xdsl-opt %s --parsing-diagnostics --split-input-file | filecheck %s
+
+// A correct operation
+
+builtin.module {
+  %0 = "test.op"() : () -> !test.type<"foo">
+  %1 = "test.op"(%0) : (!test.type<"foo">) -> !test.type<"bar">
+}
+
+// CHECK:      builtin.module {
+// CHECK-NEXT:   %0 = "test.op"() : () -> !test.type<"foo">
+// CHECK-NEXT:   %1 = "test.op"(%0) : (!test.type<"foo">) -> !test.type<"bar">
+// CHECK-NEXT: }
+
+// -----
+
+// An operation signature with not enough results
+
+builtin.module {
+  %0 = "test.op"() : () -> ()
+}
+// CHECK: Operation has 0 results, but were given 1 to bind.
+
+// -----
+
+// An operation signature with too many results
+
+builtin.module {
+  %0 = "test.op"() : () -> (!test.type<"foo">, !test.type<"bar">)
+}
+
```

Comment:
Isn't there a check missing here? Or do we actually alow the SSAValue packs as in MLIR?